source/extensions/filters/network/thrift_proxy/config.cc

```diff
@@ -135,6 +146,17 @@ ConfigImpl::ConfigImpl(
       processFilter(filter);
     }
   }
+
+  if (config.has_trds()) {
+    if (config.has_route_config()) {
+      throw EnvoyException("both trds and route_config is present in ThriftProxy");
+    }
+    route_config_provider_ = route_config_provider_manager.createRdsRouteConfigProvider(
+        config.trds(), context_.getServerFactoryContext(), stats_prefix_, context_.initManager());
+  } else {
+    route_config_provider_ = route_config_provider_manager.createStaticRouteConfigProvider(
```

Comment:
The else part should throw an exception if `config.route_config()` doesn't exist.

Reply:
It is intentional.
However that case is extreme, but the exception would be backward incompatible.
So if no trds, the `config.route_config()` is used just like in the original code: `new Router::RouteMatcher(config.route_config())` even if not specified.

Reply:
Thanks for clarifying. In that case I agree that it might make sense to keep it as is.
BTW: what would happen if the `route_config()` is empty (not-specified)? I assume that this field should've been mandatory in the first place, but don't have enough context.

Reply:
> BTW: what would happen if the route_config() is empty (not-specified)? 

It will give back reference to a `RouteConfiguration` object which has empty `routes` list. The thrift filter will drop all request because no cluster can be found.
The `routes` is also not mandatory just like the `virtual_hosts` at http. So even if make `route_config` required it is not enough.